src/Common/tests/System/Net/Configuration.Certificates.cs

```diff
@@ -45,14 +45,24 @@ private static X509Certificate2Collection GetCertificateCollection(string certif
             {
                 // On Windows, .Net Core applications should not import PFX files in parallel to avoid a known system-level race condition.
                 // This bug results in corrupting the X509Certificate2 certificate state.
-                Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
-
-                var certCollection = new X509Certificate2Collection();
-                certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
+                try
+                {
+                    Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
 
-                m.ReleaseMutex();
+                    var certCollection = new X509Certificate2Collection();
+                    certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
```

Comment:
If this throws, I assume the `Debug.Fail()` would crash the process on debug builds, in which case it would not call `ReleaseMutex()` and that may in turn unnecessarily cause `AbandonedMutexException` to be thrown in a parallel run (if the mutex happens to be opened in a parallel run as well).

If you want to use `Debug.Fail`, I would suggest something like this:
```c#
try
{
    // Assert.True(WaitOne)
}
catch(AbandonedMutexException)
{
}

try
{
    try
    {
        // Locked code
    }
    finally
    {
        // ReleaseMutex
    }
}
catch(Exception ex)
{
    // Debug.Fail
}
```